### PR TITLE
Fix git lfs path

### PR DESCRIPTION
**Pull request by sapk**
After re-testing https://github.com/go-gitea/gitea/pull/2938 and relooking at specs.

Found two errors from me : 
 - Specs indicate : https://github.com/git-lfs/git-lfs/blob/master/docs/api/locking.md 
> All File Locking requests require the following HTTP headers:
> Accept: application/vnd.git-lfs+json
> Content-Type: application/vnd.git-lfs+json

But git client (at least git version 2.15.0) doesn't send `Content-Type: application/vnd.git-lfs+json` for GET methods. -> so remove check for it.

 - Path for LFS is `:reponame.git` not `:reponame`.
 This is also use for batch path https://github.com/git-lfs/git-lfs/blob/master/docs/api/batch.md

### CODE REVIEW

**Comment by lafriks** [comment]
LGTM

**Comment by Bwko** [comment]
LGTM

**Comment by lafriks** [comment]
@sapk you are also missing migration step to add new lock table that could cause problems later if upgrading from <1.3 to something newer where is migration step that changes lock table

**Comment by sapk** [comment]
@lafriks Shouldn't https://github.com/go-gitea/gitea/blob/master/models/models.go#L120 will automatically create the table if not exist ?

**Comment by lafriks** [comment]
@sapk yes it will but if you add migration that touches or uses table that still does not exists (like upgrading from version where such table did not exist) that migration will fail as tables are sync2'ed (created) after migrations has run

File: models/migrations/v49.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import (
+	"fmt"
+	"time"
+
+	"code.gitea.io/gitea/models"
+
+	"github.com/go-xorm/xorm"
+)
+
+func addLFSLock(x *xorm.Engine) error {
```
Comment (lafriks) [inline models/migrations/v49.go:16]: @sapk you are not calling this function from `migrations.go`

**Review by lafriks — CHANGES_REQUESTED** [review]

**Comment by sapk** [comment]
@lafriks I better understand. I remember now also problems when the struct was updated and depending of version installed previously users suffer random bug. Good catch and done.

**Comment by sapk** [comment]
I need sleep ^^. Should be good. 
Sorry for the mistakes.

**Review by lafriks — APPROVED** [review]

**Pull request merged**